### PR TITLE
Score p

### PR DESCRIPTION
**Pull request by nvarini**

### CODE REVIEW

File: var/spack/repos/builtin/packages/scorep/package.py [hunk 1]
```diff
@@ -34,18 +34,20 @@ class Scorep(AutotoolsPackage):
...
 
+    version('4.1',   '3acad79cedd5c8c1456b0a8521760ba5')
     version('3.1',   '065bf8eb08398e8146c895718ddb9145')
     version('3.0',   '44da8beaa3f71436a5f6fe51938aab2f')
     version('2.0.2', '8f00e79e1b5b96e511c5ebecd10b2888')
     version('1.4.2', '3b9a042b13bdd5836452354e6567f71e')
     version('1.3',   '9db6f957b7f51fa01377a9537867a55c')
 
-    patch('gcc7.patch')
+    patch('gcc7.patch',when='gcc%7i:')
```
Comment (rmsds) [inline var/spack/repos/builtin/packages/scorep/package.py]: Not sure about the syntax here for the `when=`, did you mean this?
```suggestion
    patch('gcc7.patch', when='%gcc@7:')
```
(A space was missing a space which makes flake8 unhappy.)

File: var/spack/repos/builtin/packages/scorep/package.py [hunk 2]
```diff
@@ -34,18 +34,20 @@ class Scorep(AutotoolsPackage):
...
 
     ##########
     # Dependencies for SCORE-P are quite tight. See the homepage for more
     # information.
-    # SCOREP 3
+    # SCOREP 3 and 4
+    depends_on('cube')
```
Comment (rmsds) [inline var/spack/repos/builtin/packages/scorep/package.py]: If it's under the above comment should this be?
```suggestion
    depends_on('cube', when='@3:')
```
Or is it a general dependency for all scorep versions?

**Review by rmsds — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/scorep/package.py [hunk 3]
```diff
@@ -34,18 +34,20 @@ class Scorep(AutotoolsPackage):
...
 
+    version('4.1',   '3acad79cedd5c8c1456b0a8521760ba5')
     version('3.1',   '065bf8eb08398e8146c895718ddb9145')
     version('3.0',   '44da8beaa3f71436a5f6fe51938aab2f')
     version('2.0.2', '8f00e79e1b5b96e511c5ebecd10b2888')
     version('1.4.2', '3b9a042b13bdd5836452354e6567f71e')
     version('1.3',   '9db6f957b7f51fa01377a9537867a55c')
 
-    patch('gcc7.patch')
+    patch('gcc7.patch', when='gcc%7:')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/scorep/package.py]: I think there's an error here: it should be `when='%gcc@7:'` i.e. it should be using `@` and not `%`.

**Review by alalazo — CHANGES_REQUESTED** [review]
There's a typo in one of the constraints.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
Apparently Travis went green, but the notification to Github wasn't pushed. Waiting for @rmsds to check and eventually merge.

**Review by rmsds — APPROVED** [review]

**Pull request merged**